### PR TITLE
fix(rbac): add permission checks in api and use owner id for provider sync

### PR DESCRIPTION
**Pull request by HzaRashid**
- add pre-flight permission checks in the api for the acting user
- pass the owner id to the synchronization logic so that provider sync can succeed for readers (actors) that do not necessarily own the deployment resources or have the necessary write or delete permissions needed for synchronization
- pass both the owner id and actor id to the crud layer to perform any resource pre-filtering enforced by the auth layer

## Summary by CodeRabbit

* **New Features**
  * Added batched flow deployment permission checks with improved audit coverage.
  * Deployment listings now consistently respect ownership and visibility rules.
* **Bug Fixes**
  * Improved deployment creation, updates, deletion, and rollback authorization.
  * Fixed cross-owner deployment synchronization and attachment counting.
  * Strengthened deletion handling when database results are uncertain.
* **Tests**
  * Expanded coverage for authorization, ownership scoping, synchronization, filtering, and deletion edge cases.

### CODE REVIEW

File: src/backend/base/langflow/services/authorization/guards.py [hunk 1]
```diff
@@ -437,6 +439,125 @@ async def ensure_flow_permission(
...
+        flow_results=[
+            (flow_id, _audit.AUDIT_ALLOW if allowed else _audit.AUDIT_DENY)
+            for flow_id, allowed in zip(flow_ids, results, strict=True)
+        ],
+    )
+    if not all(results):
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail=_DEFAULT_DENY_DETAIL,
+        )
```
[the remark this answers is not in the document]
Comment (erichare) [reply src/backend/base/langflow/services/authorization/guards.py:591]: Addressed in e074f82df3. The batch guard now validates the result count before strict zip, logs and audits a deny with expected/actual counts, and returns the documented 403 on contract mismatch. Added a regression test covering the malformed plugin result.

File: src/backend/base/langflow/services/database/models/deployment/crud.py [hunk 2]
```diff
@@ -579,7 +624,7 @@ async def delete_deployment_by_resource_key(
     user_id: UUID,
     deployment_provider_account_id: UUID,
     resource_key: str,
-) -> int:
+) -> int | UnconfirmedDeleteRowcount:
```
[the remark this answers is not in the document]
Comment (erichare) [reply src/backend/base/langflow/services/database/models/deployment/crud.py:657]: Addressed in e074f82df3. The provider-sync path now checks UNCONFIRMED_DELETE_ROWCOUNT explicitly and emits a warning with the sync scope, provider, and affected deployment IDs. Added a regression test verifying the undetermined outcome is surfaced.

File: src/backend/tests/unit/services/database/test_deployment_crud_authz_prefilter.py [hunk 3]
```diff
@@ -101,6 +103,36 @@ async def test_list_deployments_page_allowed_ids_unions_owner_and_visible(async_
     assert foreign_hidden.id not in returned
 
 
+@pytest.mark.asyncio
+async def test_list_deployments_page_allowed_ids_skips_owner_when_override_disabled(
+    async_session: AsyncSession, monkeypatch
+):
```
[the remark this answers is not in the document]
Comment (erichare) [reply src/backend/tests/unit/services/database/test_deployment_crud_authz_prefilter.py]: Addressed in e074f82df3. Removed both redundant pytest.mark.asyncio decorators; pytest auto mode continues to discover both async tests.

**Comment by erichare** [comment]
Addressed the remaining CodeRabbit review-summary nitpick in e074f82df3: TestFlowIdsForVersionIds now uses a real AsyncSession with persisted User, Flow, and FlowVersion rows for empty-input, partial-match/404, and shared-flow deduplication coverage.

Also addressed all three inline findings and resolved their threads:
- fail closed with an audited 403 when batch_enforce returns the wrong result count;
- explicitly warn when a stale deployment batch delete has an unconfirmed rowcount;
- remove the two redundant pytest.mark.asyncio decorators.

Additionally fixed the external-access ceiling ordering so flow:deploy remains admin-only when AUTHZ_ENABLED is false. Merge conflicts with release-1.12.0 are resolved. Validation: Ruff clean; 172 affected backend tests passed.

**Review by erichare — APPROVED** [review]
Pushed some small fixes and rebased. LGTM!
